File: Lista-15/go/15.go
package main
import (
    "errors"
    "fmt"
)

type ContaBancaria struct {
    Saldo float64
}

func (c *ContaBancaria) Sacar(valor float64) error {
    if valor > c.Saldo {
        return errors.New("Saldo insuficiente")
    }
    c.Saldo -= valor
    return nil
}

func main() {
    conta := ContaBancaria{100}
    err := conta.Sacar(200)
    if err != nil {
        fmt.Println(err)
    }
}

File: Lista-15/python/15.py
class SaldoInsuficienteException(Exception):
    pass

class ContaBancaria:
    def __init__(self, saldo):
        self.saldo = saldo

    def sacar(self, valor):
        if valor > self.saldo:
            raise SaldoInsuficienteException("Saldo insuficiente")
        self.saldo -= valor

# Exemplo de uso
conta = ContaBancaria(100)
try:
    conta.sacar(200)
except SaldoInsuficienteException as e:
    print(e)
